### PR TITLE
Add Valleysoft.DockerCredsProvider package

### PR DESCRIPTION
**Pull request by mthalman**
Fixes dotnet/source-build#3245

### CODE REVIEW

**Review by crummel — APPROVED** [review]

File: repos/docker-creds-provider.proj [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <PropertyGroup>
+      <BuildCommandArgs>$(ProjectDirectory)/src/Valleysoft.DockerCredsProvider/Valleysoft.DockerCredsProvider.csproj</BuildCommandArgs>
+      <BuildCommandArgs>$(BuildCommandArgs) /p:Configuration=$(Configuration)</BuildCommandArgs>
+      <BuildCommandArgs>$(BuildCommandArgs) /v:$(LogVerbosity)</BuildCommandArgs>
+      <BuildCommandArgs>$(BuildCommandArgs) $(RedirectRepoOutputToLog)</BuildCommandArgs>
+      <BuildCommandArgs>$(BuildCommandArgs) /p:Version=$(DockerCredsProviderPackageVersion)</BuildCommandArgs>
+      <BuildCommandArgs>$(BuildCommandArgs) /p:TargetFrameworks=$(NetCurrent)</BuildCommandArgs>
+    </PropertyGroup>
+
+    <Exec Command="$(DotnetToolCommand) restore /bl:restore.binlog $(BuildCommandArgs)  " EnvironmentVariables="@(EnvironmentVariables)" WorkingDirectory="$(ProjectDirectory)" IgnoreStandardErrorWarningFormat="true" />
```
Comment (MichaelSimons) [inline repos/docker-creds-provider.proj:23]: Is this space intentional?  Seems inconsistent with the other commands.
```suggestion
    <Exec Command="$(DotnetToolCommand) restore /bl:restore.binlog $(BuildCommandArgs)" EnvironmentVariables="@(EnvironmentVariables)" WorkingDirectory="$(ProjectDirectory)" IgnoreStandardErrorWarningFormat="true" />
```

Comment (MichaelSimons) [reply repos/docker-creds-provider.proj:23]: I realize this was probably copied, put given the line wrapping I see in GH, consider formating so each element is on a separate line.

**Review by MichaelSimons — APPROVED** [review]
Some minor formatting comments the rest LGTM

**Comment by mthalman** [comment]
@MichaelSimons - There's a build failure due to prebuilts:

```
Package IDs are:
System.Text.Encodings.Web.6.0.0
System.Text.Json.6.0.0
```

What is the recommended solution for external repos like this?

**Comment by mthalman** [comment]
> The piece that needs validation that may not be possible yet, is that when the SDK consumes DockerCredsProvider, it picks up the versions of these that ship with the product. If they package up these SBRPs that would obviously be an issue. If that were the case, then we would need to use "current" version.

Hopefully it would never be the case where the SDK ships an older version of `System.Text.Json`, for example. That just seems like a bug to me and not something that needs to be compensated for.

File: repo-projects/docker-creds-provider.proj [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<Project>
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <PackagesOutput>$(ProjectDirectory)/src/Valleysoft.DockerCredsProvider/bin/$(Configuration)/</PackagesOutput>
+    <RepoApiImplemented>false</RepoApiImplemented>
```
Comment (MichaelSimons) [inline repo-projects/docker-creds-provider.proj:6]: @mthalman, Can you apply the cleanup from https://github.com/dotnet/source-build-externals/pull/115 to this PR?

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**